x/staking/client/cli/tx.go

```diff
@@ -33,7 +34,7 @@ var (
 )
 
 // NewTxCmd returns a root CLI command handler for all x/staking transaction commands.
-func NewTxCmd() *cobra.Command {
+func NewTxCmd(valAddrCodec, ac address.Codec) *cobra.Command {
```

Comment:
did you explore another why to provide these codecs as dependencies beside constructor injection? another approach which comes to mind is packing them into the command context.

Reply:
@tac0turtle this actually brings a good point…

Reply:
yup agreed